src/axom/core/memory_management.hpp

```diff
@@ -22,6 +22,13 @@
 namespace axom
 {
 
+#ifdef AXOM_USE_UMPIRE
+const int DEFAULT_ALLOCATOR_ID =
+  umpire::ResourceManager::getInstance().getAllocator("HOST").getId();
```

Comment:
I see this is in a header, but do you have to worry about when this is initialized? 
Could this cause a static initialization issue in the future?

Reply:
We probably have to think about this one a little more carefully. 

@davidbeckingsale can you weigh in here? 

Can we always assume that the ID of the `HOST` allocator in `Umpire` will always be `0`? I'd rather just have a `constexpr int DEFAULT_ALLOCATOR_ID=0`?

@cyrush  If this is `constexpr` do you foresee any problems down the road with this?

Reply:
Umpire objects are static initialization safe. You can call that function there.

We do not guarantee allocator IDs.  IDs of the default allocators can (and have) changed due to us having to add additional internal allocators.

I would suggest exposing the allocator name rather than an ID.

Reply:
Sounds good. Thanks for the clarification @davidbeckingsale!